### PR TITLE
Merge test mods together, Create test mods for each MC version and clean up dummy sub modules

### PR DESCRIPTION
**Pull request by thecatcore**

### CODE REVIEW

**Review by BluCobalt — APPROVED** [review]

File: legacyfabric-api/1.8/src/testmod/java/net/legacyfabric/fabric/test/client/rendering/ModelFeatureTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+package net.legacyfabric.fabric.test.client.rendering;
+
+import net.fabricmc.api.ClientModInitializer;
+
+public class ModelFeatureTest implements ClientModInitializer {
+	@Override
+	public void onInitializeClient() {
+		//		LivingEntityFeatureRendererRegistrationCallback.EVENT.register((entityClass, entityRenderer, registrationHelper) -> {
```
Comment (BoogieMonster1O1) [inline legacyfabric-api/1.8/src/testmod/java/net/legacyfabric/fabric/test/client/rendering/ModelFeatureTest.java:25]: what's up with this?

**Comment by BoogieMonster1O1** [comment]
there seem to be a lot of duplicated testmod code. ideally it could be just like how the api modules are - a common sourceset for and individual versioned sourcesets. could go in another PR too

File: legacyfabric-api/1.8/src/testmod/java/net/legacyfabric/fabric/test/client/rendering/ModelFeatureTest.java (hunk 1 above)
Comment (thecatcore) [reply legacyfabric-api/1.8/src/testmod/java/net/legacyfabric/fabric/test/client/rendering/ModelFeatureTest.java:25]: 1.8 have generics stripped. The api is built against 1.8.9 with generics, can't use in 1.8 where they don't exist which somehow we never caught on before now.

Comment (BoogieMonster1O1) [reply legacyfabric-api/1.8/src/testmod/java/net/legacyfabric/fabric/test/client/rendering/ModelFeatureTest.java:25]: you can delete this then

Comment (thecatcore) [reply legacyfabric-api/1.8/src/testmod/java/net/legacyfabric/fabric/test/client/rendering/ModelFeatureTest.java]: Done

**Comment by thecatcore** [comment]
> there seems to be a lot of duplicated testmod code. ideally it could be just like how the api modules are - a common sourceset for and individual versioned sourcesets. could go in another PR too

This will go in another PR. This is just a first step to improve test-mods, there are still more to come.

**Pull request merged**